### DISCUSSION TITLE
Lesson 9 Smart Contract Lottery ERROR

### DISCUSSION THREAD

**Discussion by yungdvnial** [Cyfrin/foundry-full-course-cu#1511] [Q&A] [answered]
Hey guys, im facing a hard time debugging this project. I'm at the point where everything is 99% completed, but when I tried to test by running:

(forge test --fork-url $SEPOLIA_RPC_URL -vvvv)

it returns an error. And when i tried to deploy it straight on the sepolia testnet by running:

(forge script script/DeployRaffle.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $SEPOLIA_PRIVATE_KEY --verify --etherscan-api-key $ETHERSCAN_API_KEY )

it also returns the same error. I do not understand why this happening. 

Here is the snapshot of the error: 
<img width="1089" alt="Screenshot 2024-03-22 at 6 01 15 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/145892850/ab37fa96-415d-4cda-986d-8d4f10e34398">

Here is the AddConsumer function within interactions.s.sol:
<img width="563" alt="Screenshot 2024-03-22 at 6 05 57 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/145892850/de7d3dce-c14c-4aba-89ba-da5f49860a8c">

Here is the AddConsumer syntax within deploy.s.sol:
<img width="373" alt="Screenshot 2024-03-22 at 6 07 29 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/145892850/1c351b96-a917-44fa-8eb5-de5de2ee498b">

Here is the Sepolia function from helperconfig.s.sol:
<img width="686" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/145892850/73e01b5d-ae89-49b7-b099-53f87cda779c">

Here's my [repo](https://github.com/everybodylovesdanial/Solidity-Smart-Contract-Lottery) for you guys to check.

I would really appreciate it if you could help, thanks :)

**Comment by EngrPips** [comment]
Hello @everybodylovesdanial , i really appreciate that you shared your repo already, i will set your repo locally, try debug it and come back with whatever i find.

**Reply by yungdvnial** [reply]
Hey @EngrPips, heres my repo [smart-contract-lottery](https://github.com/everybodylovesdanial/Solidity-Smart-Contract-Lottery)

**Comment by EngrPips** [comment, marked as answer]
Hello @everybodylovesdanial , below is a picture of the result of running your test on my local machine

<img width="1440" alt="Screenshot 2024-03-22 at 3 00 01 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/45576415-6f6e-4996-a04c-367f224a5436">

As you can see every single test passed and that lead me to the question about the `subId` you used in your `HelperConfig.sol` for the `getSepoliaEthConfig` function. Because it seem as if that `subId` that you used is no way related to the `private key` you passed in as the `deployerKey` in the `getSepoliaEthConfig` function and for that reason you are getting the error `MustBeSubOwner`

Looking forward to hear from you.

**Reply by yungdvnial** [reply]
Yep it was the private key. That issue is solved now and the tests went smoothly. However, I encountered another issue while deploying on the real testnet:

After running forge script script/DeployRaffle.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $SEPOLIA_PRIVATE_KEY --verify --etherscan-api-key $ETHERSCAN_API_KEY:
<img width="967" alt="Screenshot 2024-03-22 at 11 21 41 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/145892850/519e73c6-3262-4b66-b786-8037f20245e0">

**Reply by EngrPips** [reply]
Hello @everybodylovesdanial , copy and paste the command below in your terminal and your contract should successfully deploy on sepolia 

```
forge script script/DeployRaffle.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $SEPOLIA_PRIVATE_KEY --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY
```

**Reply by yungdvnial** [reply]
works now. Thanks !!!

**Reply by EngrPips** [reply]
I am glad it worked. keep crushing it champ 💪
